src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/NuGet/PackageRestoreUnconfiguredDataSource.DropConfiguredProjectVersionDataSource.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks.Dataflow;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.NuGet
+{
+    internal partial class PackageRestoreUnconfiguredDataSource
+    {
+        /// <summary>
+        ///     A <see cref="IProjectValueDataSource{T}"/> that drops <see cref="ProjectDataSources.ConfiguredProjectIdentity"/> and 
+        ///     <see cref="ProjectDataSources.ConfiguredProjectVersion"/> versions from each value of the original
+        ///     <see cref="IProjectValueDataSource{T}"/>.
+        /// </summary>
+        private sealed class DropConfiguredProjectVersionDataSource<T> : ChainedProjectValueDataSourceBase<T>
```

Comment:
We have a cheaper implementation (TransformLinkTo inside CPS), but haven't made it public. For a minor version transform, it makes sense to use that one (instead of creating a full size dataflow block).  Let me know if you need, and we can update our side.